src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -16,19 +16,34 @@ namespace Microsoft.PowerShell.Commands
     /// <summary>
     /// This class implements Test-Json command.
     /// </summary>
-    [Cmdlet(VerbsDiagnostic.Test, "Json", DefaultParameterSetName = ParameterAttribute.AllParameterSets, HelpUri = "https://go.microsoft.com/fwlink/?LinkID=2096609")]
+    [Cmdlet(VerbsDiagnostic.Test, "Json", DefaultParameterSetName = JsonStringParameterSet, HelpUri = "https://go.microsoft.com/fwlink/?LinkID=2096609")]
     [OutputType(typeof(bool))]
     public class TestJsonCommand : PSCmdlet
     {
-        private const string SchemaFileParameterSet = "SchemaFile";
-        private const string SchemaStringParameterSet = "SchemaString";
+        private const string JsonStringParameterSet = "JsonString";
+        private const string JsonStringWithSchemaStringParameterSet = "JsonStringWithSchemaString";
+        private const string JsonStringWithSchemaFileParameterSet = "JsonStringWithSchemaFile";
+        private const string JsonFileParameterSet = "JsonFile";
+        private const string JsonFileWithSchemaStringParameterSet = "JsonFileWithSchemaString";
+        private const string JsonFileWithSchemaFileParameterSet = "JsonFileWithSchemaFile";
 
         /// <summary>
         /// Gets or sets JSON string to be validated.
         /// </summary>
-        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true)]
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true, ParameterSetName = JsonStringParameterSet)]
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true, ParameterSetName = JsonStringWithSchemaStringParameterSet)]
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true, ParameterSetName = JsonStringWithSchemaFileParameterSet)]
         public string Json { get; set; }
 
+        /// <summary>
+        /// Gets or sets JSON file path to be validated.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = JsonFileParameterSet)]
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = JsonFileWithSchemaStringParameterSet)]
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = JsonFileWithSchemaFileParameterSet)]
+        [Alias("JsonFile")]
+        public string Path { get; set; }
```

Comment:
It is not clear should the parameter work as Path or LiteralPath? Works with pipeline?

Reply:
@iSazonov Not sure how that would work, given `Json` parameter is already value from pipeline. Would it be beneficial to include a `LiteralPath` parameter as well. To keep consistent with `Get-ChildItem`.

Reply:
Or maybe other way around.

Reply:
Might make sense to just keep `Path` and use `SessionState.Path.GetUnresolvedProviderPathFromPSPath`.

Reply:
I believe this is how the `Add-Type` cmdlet works. Could be worthwhile including both parameters, although then would need to think about if the parameters need to be treated as arrays or not.

Reply:
"> @iSazonov Not sure how that would work, given `Json` parameter is already value from pipeline.

Need consider ValueFromPipelineByPropertyName

As for "Path and/or LiteralPath", we have different patterns in some cmdlets. We need a conclusion in related issue.